crates/stages/src/stages/merkle.rs

```diff
@@ -70,15 +127,411 @@ impl<DB: Database> Stage<DB> for MerkleStage {
     /// Unwind the stage.
     async fn unwind(
         &mut self,
-        _tx: &mut Transaction<'_, DB>,
+        tx: &mut Transaction<'_, DB>,
         input: UnwindInput,
     ) -> Result<UnwindOutput, StageError> {
-        if matches!(self, MerkleStage::Execution) {
+        if matches!(self, MerkleStage::Execution { .. }) {
             info!(target: "sync::stages::merkle::exec", "Stage is always skipped");
             return Ok(UnwindOutput { stage_progress: input.unwind_to })
         }
+        let loader = DBTrieLoader::default();
+        let current_root = tx.get_header_by_num(input.stage_progress)?.state_root;
+
+        let from_transition = tx.get_block_transition(input.unwind_to)? + 1;
+        let to_transition = tx.get_block_transition(input.stage_progress)? + 1;
```

Comment:
It needs to be without `+1` there was small miscalculation earlier that was fixed on `main`

Reply:
More in depth explanation: https://github.com/paradigmxyz/reth/pull/894#discussion_r1071600828

Reply:
Tests fail by changing that. What was the miscalculation? As I understand, `get_block_transition` returns the last transition id of the block, and by adding one we get the first of the next block.

Reply:
It is for the initial case to start changesets from index zero, and it is easier to think about taking `[from,to)` where `from` is inclusive and `to` is exclusive will not e processed. 
And if there are no transactions this `+1` can make a footgun gap in transition index.

Here is a rust example that you can play with: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=ea0f8335487dc0ccae4519cb0abd2294 
I needed to make it to wrap my head around it :D

Reply:
cc: https://github.com/paradigmxyz/reth/pull/994/files#r1092238031